### PR TITLE
logging: structured logging, klog deprecation

### PR DESCRIPTION
**Pull request by pohly**
This is primarily the docs PR for
- alpha: https://github.com/kubernetes/enhancements/issues/2845
- beta: https://github.com/kubernetes/enhancements/issues/1602
    
But as it touches the file, it also updates the examples and explanations to
reflect some recent observations.

### CODE REVIEW

File: content/en/docs/concepts/cluster-administration/system-logs.md [hunk 1]
```diff
@@ -22,11 +22,59 @@ generates log messages for the Kubernetes system components.
 
 For more information about klog configuration, see the [Command line tool reference](/docs/reference/command-line-tools-reference/).
```
Comment (pohly) [inline content/en/docs/concepts/cluster-administration/system-logs.md:23]: What is the right approach for updating the command line tools reference? Manually edit the HTML table?

That needs to be part of this PR.

/hold

Comment (sftim) [reply content/en/docs/concepts/cluster-administration/system-logs.md:23]: We typically update all [generated](https://github.com/kubernetes-sigs/reference-docs) docs during the release, separately from human-made PRs that accompany changes in that release.

If that's true here to, we can /unhold.

File: content/en/docs/concepts/cluster-administration/system-logs.md [hunk 2]
```diff
@@ -22,11 +22,59 @@ generates log messages for the Kubernetes system components.
...
+node release archives.
+
+This table shows how `kube-log-runner` invocations correspond to shell redirection:
+
+| Usage                                    | POSIX shell (such as bash) | `kube-log-runner <options> <cmd>`                           |
+| -----------------------------------------|----------------------------|-------------------------------------------------------------|
+| Merge stderr and stdout, write to stdout | `2>&1`                     | `kube-log-runner` (default behavior)                        |
+| Redirect both into log file              | `1>>/tmp/log 2>&1`         | `kube-log-runner -log-file=/tmp/log`                        |
+| Copy into log file and to stdout         | `2>&1 \| tee -a /tmp/log`  | `kube-log-runner -log-file=/tmp/log -also-stdout`           |
+| Redirect only stdout into log file       | `>/tmp/log`                | `kube-log-runner -log-file=/tmp/log -redirect-stderr=false` |
```
[the remark this answers is not in the document]
Comment (sftim) [reply content/en/docs/concepts/cluster-administration/system-logs.md:62]: Here's the preview: https://deploy-preview-30505--kubernetes-io-vnext-staging.netlify.app/docs/concepts/cluster-administration/system-logs/#klog

(To find this: start from https://github.com/kubernetes/website/pull/30505#issuecomment-970379154 and click “Deploy preview”)

File: content/en/docs/concepts/cluster-administration/system-logs.md (hunk 1 above)
Comment (pohly) [reply content/en/docs/concepts/cluster-administration/system-logs.md:23]: Sounds like I don't need to worry about the reference docs then.

/unhold

File: content/en/docs/concepts/cluster-administration/system-logs.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (pohly) [reply content/en/docs/concepts/cluster-administration/system-logs.md:62]: Hmm, "Deploy preview" is just plain text when I look at that comment.

I see a link to the deploy log, but that also doesn't link to the preview.

Anyway, rendering looks good.

[the remark this answers is not in the document]
Comment (sftim) [reply content/en/docs/concepts/cluster-administration/system-logs.md:62]: Ah sorry. It turns out that as a privileged user I get that link and other folks don't.
Not sure why (I don't see it as confidential and the URLs are predictable).

I might nudge Netlify about that if I get time. It's a busy period though!

**Comment by serathius** [comment]
LG from technical standpoint. 
/lgtm

**Review by mehabhalodiya — COMMENTED** [review]
LGTM from SIG Docs!

**Comment by pohly** [comment]
/assign @reylejano

For approval.

**Pull request merged**